src/extractor/guidance/turn_handler.cpp

```diff
@@ -87,48 +142,37 @@ Intersection TurnHandler::handleTwoWayTurn(const EdgeID via_edge, Intersection i
 }
 
+// checks whether it is obvious to turn on `road` coming from `via_edge` while there is an`other`
+// road at the same intersection
 bool TurnHandler::isObviousOfTwo(const EdgeID via_edge,
                                  const ConnectedRoad &road,
                                  const ConnectedRoad &other) const
 {
-    const auto &in_data = node_based_graph.GetEdgeData(via_edge);
-
-    const auto &first_data = node_based_graph.GetEdgeData(road.eid);
-    const auto &second_data = node_based_graph.GetEdgeData(other.eid);
-    const auto &first_classification = first_data.road_classification;
-    const auto &second_classification = second_data.road_classification;
-    const bool is_ramp = first_classification.IsRampClass();
-    const bool is_obvious_by_road_class =
-        (!is_ramp &&
-         (2 * first_classification.GetPriority() < second_classification.GetPriority()) &&
-         in_data.road_classification == first_classification) ||
-        (!first_classification.IsLowPriorityRoadClass() &&
-         second_classification.IsLowPriorityRoadClass());
-
-    if (is_obvious_by_road_class)
-        return true;
-
-    const bool other_is_obvious_by_road_class =
-        (!second_classification.IsRampClass() &&
-         (2 * second_classification.GetPriority() < first_classification.GetPriority()) &&
-         in_data.road_classification == second_classification) ||
-        (!second_classification.IsLowPriorityRoadClass() &&
-         first_classification.IsLowPriorityRoadClass());
+    const auto &via_data = node_based_graph.GetEdgeData(via_edge);
+    const auto &road_data = node_based_graph.GetEdgeData(road.eid);
+    const auto &other_data = node_based_graph.GetEdgeData(other.eid);
+    const auto &via_classification = via_data.road_classification;
+    const auto &road_classification = road_data.road_classification;
+    const auto &other_classification = other_data.road_classification;
 
-    if (other_is_obvious_by_road_class)
+    // if one of the given roads is obvious by class, obviousness is trivial
+    if (obviousByRoadClass(via_classification, road_classification, other_classification))
+    {
+        return true;
+    }
+    else if (obviousByRoadClass(via_classification, other_classification, road_classification))
+    {
         return false;
+    }
 
     const bool turn_is_perfectly_straight =
         angularDeviation(road.angle, STRAIGHT_ANGLE) < std::numeric_limits<double>::epsilon();
-
-    const auto &road_data = node_based_graph.GetEdgeData(road.eid);
-
     const auto same_name = !util::guidance::requiresNameAnnounced(
-        in_data.name_id, road_data.name_id, name_table, street_name_suffix_table);
-
-    if (turn_is_perfectly_straight && in_data.name_id != EMPTY_NAMEID &&
-        road_data.name_id != EMPTY_NAMEID && same_name)
+        via_data.name_id, road_data.name_id, name_table, street_name_suffix_table);
+    if (turn_is_perfectly_straight && via_data.name_id != EMPTY_NAMEID && same_name)
```

Comment:
Wait, the empty name id check has to happen _before_ the `requiresNameAnnounced` above